lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ServiceAndContextConfigurationProvider.cs

```diff
@@ -69,113 +66,245 @@ internal sealed class ServiceAndContextConfigurationProvider<TMapInput, TMapOutp
         /// submitted evaluator to failed evaluator
         /// </summary>
         /// <param name="evaluatorId"></param>
-        internal void EvaluatorFailed(string evaluatorId)
+        /// <returns>Whether failed evaluator is master or not</returns>
```

Comment:
I think this class tries to do way too many things, would be easier to understand if it was broken up -- though I'm not sure if it's a task for this PR.